### PR TITLE
MGMT-4694 Adds KubeAPI CRD examples

### PR DESCRIPTION
**Pull request by nmagnezi**
This will replace https://github.com/nmagnezi/assisted-service-crs

### CODE REVIEW

File: docs/Kube-API-examples.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#Assisted Installer Kube API CR examples
+
+[docs/crds](https://github.com/openshift/assisted-service/tree/master/docs/crds) stores working examples of various resources we spawn via kube-api in assisted-installer, for Hive intergration.
+Those examples are here for reference.
+
+You will likely need to adapt those for your own needs.
+
```
Comment (nmagnezi) [inline docs/Kube-API-examples.md:7]: for reviewers: note that those URLs will only work after this PR gets merged.

File: docs/crds/installEnv.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    namespace: assisted-installer
+  agentLabelSelector:
+    matchLabels:
+      bla: aaa
+  pullSecretRef:
+    name: pull-secret
+  proxy:
+    httpProxy: http://11.11.11.33
+    httpsProxy: http://22.22.22.55
+  ignitionConfigOverride: '{"ignition": {"version": "3.1.0"}, "storage": {"files": [{"path": "/tmp/example", "contents": {"source": "data:text/plain;base64,aGVscGltdHJhcHBlZGluYXN3YWdnZXJzcGVj"}}]}}'
```
Comment (rollandf) [inline docs/crds/installEnv.yaml:17]: /tmp/example -> etc/someconfig

File: docs/crds/clusterDeployment.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      name: mycluster-install-config
+    sshPrivateKeySecretRef:
+      name: mycluster-ssh-key
+    installStrategy:
+      agent:
+        networking:
+          clusterNetwork:
+            - cidr: 10.128.0.0/14
+              hostPrefix: 23
+          ServiceNetwork:
```
[the remark this answers is not in the document]
Comment (nmagnezi) [reply docs/crds/clusterDeployment.yaml]: Done

File: docs/Kube-API-examples.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#Assisted Installer Kube API CR examples
+
+[docs/crds](https://github.com/openshift/assisted-service/tree/master/docs/crds) stores working examples of various resources we spawn via kube-api in assisted-installer, for Hive intergration.
```
[the remark this answers is not in the document]
Comment (nmagnezi) [reply docs/Kube-API-examples.md]: Done

File: docs/crds/installEnv.yaml (hunk 2 above)
Comment (nmagnezi) [reply docs/crds/installEnv.yaml]: Done

**Pull request merged**